inline.h

```diff
@@ -57,6 +57,98 @@ Perl_av_count(pTHX_ AV *av)
     return AvFILL(av) + 1;
 }
 
+/* ------------------------------- av.c ------------------------------- */
+
+/*
+=for apidoc av_store_simple
+
+This is a cut-down version of av_store that assumes that the array is
+very straightforward - no magic, not readonly, and AvREAL - and that
+C<key> is not negative. This function MUST NOT be used in situations
+where any of those assumptions may not hold.
+
+If any of those conditions actually turns out
+to be true, this function will punt to av_store. (Assuming that the
+checks haven't been optimized away due to ASSUME()s at the call site.)
```

Comment:
This is no longer true.

Reply:
Thanks Tony. I've pushed an update deleting that comment block.